src/SchemaGenerator/ObjectSchemaGenerator.php

```diff
@@ -120,7 +120,16 @@ public function generate(\ReflectionClass $reflectionClass, array $context = [])
         }
 
         $classMetadata = $this->registry->getManagerForClass($className)->getClassMetadata($className);
-        foreach ($this->propertyInfo->getProperties($className, $context) as $property) {
+        $classProperties = $this->propertyInfo->getProperties($className, $context);
+
+        if (empty($classProperties) && 1 !== $context['depth']) {
```

Comment:
Why `1 !== $context['depth']`?

Reply:
I had a case where I needed that, i'll get you in touch when I find it again